### ISSUE TITLE
Strings are immutable now

### ISSUE THREAD

**Issue by prekel** [realworldocaml/book#3529]
TLDR: [Imperative Programming, Strings](https://dev.realworldocaml.org/imperative-programming.html#strings) says that strings are mutable

I was very surprised when I read [Imperative Programming, Strings](https://dev.realworldocaml.org/imperative-programming.html#strings) because I thought that strings in OCaml are immutable, but didn't check it in `utop` and left it as an interesting fact. But after reading [Jérôme Vouillon's and Vincent Balat's "From Bytecode to Javascript: the Js_of_ocaml Compiler"](https://www.irif.fr/~balat/publications/vouillon_balat-js_of_ocaml.pdf) I find out that JavaScript strings are immutable and therefore `js_of_ocaml` has their own implementation of strings. And afterward, I realize that ReScript uses JS's string representation. While investigating it, I found that there were [changes](https://github.com/ocaml/ocaml/blob/trunk/Changes) in the OCaml compiler after that fragment was written which make strings immutable by default:
> - OCaml 4.02.0 (29 Aug 2014): Separation between read-only strings (type string) and read-write byte sequences (type bytes). Activated by command-line option -safe-string.
> - OCaml 4.06.0 (3 Nov 2017): https://github.com/ocaml/ocaml/pull/1252: The default mode is now safe-string, can be overridden at configure time or at compile time.
> - OCaml 4.10.0 (21 February 2020): https://github.com/ocaml/ocaml/pull/1859: enforce safe (immutable) strings by removing the -unsafe-string option by default.

So, I suggest rewriting [Imperative Programming, Strings](https://dev.realworldocaml.org/imperative-programming.html#strings) for bytes and to mention bytes in [Memory Representation of Values, String Values](https://dev.realworldocaml.org/runtime-memory-layout.html#string-values).

**Comment by yminsky** [comment]
Oh man, that's quite an oversight. Thanks for the catch! We'll fix it.
